tracer/src/Datadog.Trace/Iast/IastRequestContext.cs

```diff
@@ -203,14 +204,26 @@ internal void AddRequestData(System.Web.HttpRequest request)
                 helper.AddExecutedSource(IastInstrumentedSources.RequestUri);
             }
 
-            if (request.QueryString != null)
+            NameValueCollection? queryString = null;
+
+            try
+            {
+                // we can get an exception if the query string fails validation
+                queryString = request.QueryString;
+            }
+            catch (HttpRequestValidationException)
+            {
+                Log.Debug("Error reading request QueryString.");
+            }
```

Comment:
Yikes, really?! We should probably extract this to a helper method and make sure we use it everywhere, include in APM too!🙂

Reply:
Done!